runtime/src/accounts_db.rs

```diff
@@ -803,6 +882,14 @@ impl AccountsDB {
                                     );
                                 }
 
+                                let has_zero_lamport_accounts = locked_entry
```

Comment:
the reason of looping here is that you need to check all entries regardless `max_clean_root`? how about commenting the subtlety?

Reply:
Maybe a comment here as well that this check is sufficient, because any "outdated" zero-lamport keys, i.e. keys that became zero-lamport but then received a non-zero value at a later slot, will eventually be purged from the index by the call to `clean_old_rooted_accounts` below, and then this check will succeed on the next clean.

Reply:
Yes, the zero_lamport_pubkey index is meant to contain *all* keys which may have a 0-lamport update in them, even if it is rooted or not, because it may become rooted in the future. This would then prune any entries which do not have a 0-lamport update because it was removed in some other clean or update operation.